### PR TITLE
[Doc] CN User Guide Chapter 7

### PR DESCRIPTION
**Pull request by zhjwy9343**
## Description
Chinese User Guide Chapter 7
 
## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: docs/source/guide_cn/distributed.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+:ref:`(English Version) <guide-distributed>`
+
+DGL采用完全分布式的方法，可将数据和计算同时分布在一组计算资源中。在本节中，
+我们默认使用一个集群的环境设置(即一组机器)。DGL会将一张图划分为多张子图，
+集群中的每台机器各自负责一张子图(分区)。为了并行化计算，DGL在集群所有机器上运行相同的训练脚本，
+并在同样的机器上运行服务器以将分区数据提供给训练器。
+
+对于训练脚本，DGL提供了分布式的API。它们与小批次训练的API相似。用户仅需对单机小批次训练的代码稍作修改就可实现分布式训练。
+以下代码给出了一个用分布式方式训练GraphSage的示例。仅有的代码修改出现在第4-7行：1)初始化DGL的分布式模块，2)创建分布式图对象，以及
+3)拆分训练集，并计算本地进程的节点。其余代码保持不变，与 :ref:`mini-batch training <guide-minibatch>` 类似，
```
Comment (mufeili) [inline docs/source/guide_cn/distributed.rst:15]: "guide-minibatch" -> "guide_cn-minibatch"

File: docs/source/guide_cn/distributed.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+包括：创建采样器，模型定义，模型训练的循环。
```
Comment (mufeili) [inline docs/source/guide_cn/distributed.rst:16]: “创建采样器，模型定义，模型训练的循环” -> “创建采样器、模型定义、模型训练的循环”

File: docs/source/guide_cn/distributed.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            loss.backward()
+            optimizer.step()
+
+在一个集群的机器上运行训练脚本时，DGL提供了一些工具，可将数据复制到集群的计算机上，并在所有机器上启动训练任务。
+
+**Note**: 当前版本的分布式训练API仅支持PyTorch后端。
+
+**Note**: 当前版本的实现仅支持具有一种节点类型和一种边类型的图。
+
+DGL实现了一些分布式组件以支分布式训练，下图显示了这些组件及其它们间的相互作用。
```
Comment (mufeili) [inline docs/source/guide_cn/distributed.rst:64]: “支” -> “支持”

File: docs/source/guide_cn/distributed-preprocessing.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+它可以用最少的边分割以生成分区，从而减少了用于分布式训练和推理的网络通信。DGL使用最新版本的Metis，
+并针对真实世界中具有幂律分布的图进行了优化。在图划分后，API以易于在训练期间加载的格式构造划分结果。
+
+**Note**: 图划分API当前在一台机器上运行。 因此如果一张图很大，用户将需要一台大内存的机器来对图进行划分。
+未来DGL将支持分布式图划分。
+
+默认情况下，为了在分布式训练/推理期间定位节点/边，API将新ID分配给输入图的节点和边。
+分配ID后，该API会相应地打乱所有节点数据和边数据。在训练期间，用户只需使用新的节点和边的ID。
+与此同时，用户仍然可以通过 ``g.ndata['orig_id']`` 和 ``g.edata['orig_id']`` 获取原始ID。
+其中 ``g`` 是 ``DistGraph`` 对象（请参见7.2里的DistGraph小节）。
```
Comment (mufeili) [inline docs/source/guide_cn/distributed-preprocessing.rst:20]: “请参见7.2里的DistGraph小节”加个超链接？

**Comment by mufeili** [comment]
"cluster"我看到的翻译还是用集群的比较多。

File: docs/source/guide_cn/distributed-tools.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+:ref:`(English Version) <guide-distributed-tools>`
+
+DGL提供了两个脚本来帮助用户进行分布式训练：
+
+* *tools/copy_files.py* 用于将图分区复制到集群，
+* *tools/launch.py* 用于在机器集群中启动分布式训练任务。
+
+*copy_files.py* 将计算机(对图进行分区的计算机)中的分区数据和相关文件(例如，训练脚本)
+复制到机器集群(在其中进行分布式训练)上。在这些机器上，分布式训练将需要用到这些分区。该脚本包含四个参数：
```
Comment (mufeili) [inline docs/source/guide_cn/distributed-tools.rst:14]: “机器集群(在其中进行分布式训练)” -> “（负责分布式训练的）机器集群”

**Comment by mufeili** [comment]
Done a first pass.

File: docs/source/guide_cn/distributed-apis.rst [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    loss.backward()
+    optimizer.step()
+    sparse_optimizer.step()
+
+**Note**: :class:`~dgl.distributed.DistEmbedding` 不是PyTorch的nn模块，因此用户无法从nn模块的参数访问它。
+
+分布式采样
+~~~~~~~~~~~~~~~~~~~~
+
+DGL提供了两个级别的API，用于对节点和边进行采样以生成小批次(请参阅小批次训练的章节)。
```
Comment (aksnzhy) [inline docs/source/guide_cn/distributed-apis.rst:177]: “生成小批次训练数据”

**Comment by aksnzhy** [comment]
其他没啥问题了，翻译的很赞！

File: docs/source/guide_cn/distributed.rst (hunk 1 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed.rst:15]: changed

File: docs/source/guide_cn/distributed.rst (hunk 2 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed.rst:16]: 逗号表示顺序关系，顿号没有这个顺序。这里有一定的顺序关系，不能前后变换位置，所以还是建议用逗号。

File: docs/source/guide_cn/distributed.rst (hunk 3 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed.rst:64]: changed

[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed.rst:64]: changed

File: docs/source/guide_cn/distributed-preprocessing.rst (hunk 4 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-preprocessing.rst:20]: add links

File: docs/source/guide_cn/distributed-tools.rst (hunk 5 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-tools.rst:14]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 7]
```diff
@@ -0,0 +1,253 @@
+.. _guide_cn-distributed-apis:
+
+7.2 分布式的API
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:3]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+对于分布式训练/推断，DGL提供了三种分布式数据结构：用于分布式图的 :class:`~dgl.distributed.DistGraph`、
+用于分布式张量的 :class:`~dgl.distributed.DistTensor` 和用于分布式可学习嵌入的
+:class:`~dgl.distributed.DistEmbedding`。
+
+DGL分布式模块的初始化
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`~dgl.distributed.initialize` 可以用于初始化分布式模块。当训练脚本在训练器模式下运行时，
+这个API会与DGL服务器建立连接并创建采样器进程。当脚本在服务器模式下运行时，这个API将运行服务器代码，
+并且永不返回。必须在DGL的任何其他分布式API之前，调用此API。在使用PyTorch时，必须在
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:18]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+**Note**: :class:`~dgl.distributed.DistGraph` 当前仅支持一种节点类型和一种边类型的图。
+
+分布式模式与独立模式
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:class:`~dgl.distributed.DistGraph` 可以在两种模式下运行：分布式模式和独立模式。
+当用户在Python命令行或Jupyter Notebook中执行训练脚本时，它将以独立模式运行。也就是说，它在单个进程中运行所有计算，
+并且不与任何其他进程通信。因此，独立模式要求输入图仅具有一个分区。此模式主要用于开发和测试
+(例如，在Jupyter Notebook中开发和运行代码)。当用户使用启动脚本执行训练脚本时(请参见启动脚本部分)，
+:class:`~dgl.distributed.DistGraph` 将以分布式模式运行。启动工具在后台启动服务器(包括访问节点/边特征和图采样)，
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:46]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+    g.ndata['train_mask']
+    <dgl.distributed.dist_graph.DistTensor at 0x7fec820937b8>
+    g.ndata['train_mask'][0]
+    tensor([1], dtype=torch.uint8)
+
+分布式张量
+~~~~~~~~~~~~~~~~~
+
+如前所述，在分布式模式下，DGL会划分节点和边特征，并将它们存储在计算机集群中。
+DGL为分布式张量提供了类似于张量的接口，以访问群集中的分区节点和边特征。
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:99]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+在分布式设置中，DGL仅支持密集节点和边特征。
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:100]: changed

File: docs/source/guide_cn/distributed-apis.rst [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+但不支持一些计算运算符，例如求和以及求均值。
+
+.. code:: python
+
+    data = g.ndata['feat'][[1, 2, 3]]
+    print(data)
+    g.ndata['feat'][[3, 4, 5]] = data
+
+**Note**: 当前，当一台机器运行多个服务器时，DGL不提供对来自多个训练器的并发写入的保护。
+这可能会导致数据损坏。避免同时写入同一行数据的一种方法是在一个计算机上只运行一个服务器进程。
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:136]: changed

File: docs/source/guide_cn/distributed-apis.rst (hunk 6 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/distributed-apis.rst:177]: changed

**Pull request merged**